### PR TITLE
Allow installation of current 'structlog' major version 21.x

### PR DESCRIPTION
**Pull request by grelleum**
This PR allows using `structlog` major version 21 to be installed with `diffsync`.
This fixes issue: https://github.com/networktocode/diffsync/issues/74

Three changes are made:
1. Pinned the `pylint` version to 2.6.x so that `invoke tests` will complete successfully after running `poetry lock`.
2. Modified `structlog` versioning to allow 20.x.x or 21.x.x.  Ran complete tests to verify compatibility.
3. Updated `poetry.lock` file to include current `structlog` version.

### CODE REVIEW

**Comment by grelleum** [comment]
Thanks @dgarros - I will look further into this as see how these changes affects DiffSync.

File: diffsync/logging.py [hunk 1]
```diff
@@ -35,18 +35,52 @@ def enable_console_logging(verbosity=0):
...
+    if either rich or better_exceptions packages are available to import
+    when the 'format_exc_info' processor is used.
+
+    This code snippet will determine if we need to add 'format_exc_info'
+    to the processors.
+    """
+
+    structlog_float_version = float(".".join(structlog.__version__.split(".")[:2]))
+    if structlog_float_version < 21.2:
+        return True
```
Comment (glennmatthews) [inline diffsync/logging.py:74]: This looks pretty gross. Can we use `packaging.Version` objects here instead? Something like:

```python
from packaging import version

if version.parse(structlog.__version__) < version.Version("21.2.0"):
```

Comment (grelleum) [reply diffsync/logging.py]: Great suggestion - I've implemented the change.

File: diffsync/logging.py [hunk 2]
```diff
@@ -35,18 +38,45 @@ def enable_console_logging(verbosity=0):
...
     structlog.configure(
-        processors=[
-            structlog.stdlib.filter_by_level,  # <-- added
-            structlog.stdlib.add_logger_name,
-            structlog.stdlib.add_log_level,
-            structlog.processors.TimeStamper(fmt="%Y-%m-%d %H:%M.%S"),
-            structlog.processors.StackInfoRenderer(),
-            structlog.processors.format_exc_info,
-            structlog.dev.ConsoleRenderer(),
-        ],
```
Comment (glennmatthews) [inline diffsync/logging.py:47]: Don't we need a `processors=processors,` line added here?

**Comment by grelleum** [comment]
The structlog change in 21.2.0 and above changes behavior with regards to the use of `format_exc_info` as a log processor.
This effects exception logging when using the `logging.enable_console_logging` function.

When using 21.2.0 or newer and either `better_exceptions` or `rich` in available in the Python environment, then exception logging will generate a warning if `structlog.processors.format_exc_info` is installed as a processor.  If we remove the `format_exc_info` processor then we lose exception traceback formatting, when the above is not the case.

The code added here determines if the `structlog.processors.format_exc_info` is required in the processor list, and appends it to the list if required.

File: diffsync/logging.py (hunk 2 above)
Comment (grelleum) [reply diffsync/logging.py:47]: Indeed, it got lost somewhere in the experimentation phase.  Added this back in.

**Comment by grelleum** [comment]
For compatibility with Python 3.9, I changed:
```
import importlib
...
    rich = importlib.util.find_spec("rich")
    better_exceptions = importlib.util.find_spec("better_exceptions")
```
to
```
from importlib.util import find_spec
... 
    rich = find_spec("rich")
    better_exceptions = find_spec("better_exceptions")
```

**Review by dgarros — APPROVED** [review]
LGTM thank you @grelleum for the PR
@glennmatthews do you think we are good to merge ?

**Review by glennmatthews — APPROVED** [review]
:shipit:

**Pull request merged**